### ISSUE TITLE
Create output infoset extension matches the configured format.

### ISSUE THREAD

**Issue by hdalsania** [apache/daffodil-vscode#845]
Even if you choose output type JSON, the output infoset file is named infoset.xml. Should be infoset.json.

Source of this ticket is approval VOTE [email](https://lists.apache.org/thread/0gcddqd3mhys5179t8zgkpp51l9oc6nv) from @mbeckerle for v1.3.1-rc1 on 08/29/2023

**Comment by arosien** [comment]
#927 auto-fixes the mismatch, but I've been wondering if instead we should change the configuration so that if you want the infoset written to a file, you only get to specify the directory where the file will go, and we always name it `infoset.<extension>`. @mbeckerle @shanedell ?

**Comment by shanedell** [comment]
That could work but how do prevent users from entering a full file path. Basically if we ignore that we could get something odd looking like the user specifies the path `${workspaceFolder}/infoset.xml` but then the actual file is `${workspaceFolder}/infoset.xml/infoset.xml`. Is that something we shouldn't be concerned about or what would be a way to work around that?

**Comment by arosien** [comment]
> That could work but how do prevent users from entering a full file path. Basically if we ignore that we could get something odd looking like the user specifies the path `${workspaceFolder}/infoset.xml` but then the actual file is `${workspaceFolder}/infoset.xml/infoset.xml`. Is that something we shouldn't be concerned about or what would be a way to work around that?

Yeah that situation could happen. I think no matter what there needs to be a warning relayed to the user that they may be doing something weird, and perhaps not allowing the session to run under such a condition.

My main concern with the current PR, or an alternative, is that the error condition, and perhaps an applied fix, isn't relayed to the user. I'm ok with auto-fixing + warning, or error + stopping.
